### PR TITLE
chore: update charm pins

### PR DESCRIPTION
**Pull request by benhoyt**
This is an automated PR to update pins of the external repositories that the operator framework is tested against

### CODE REVIEW

**Review by tonyandrewmeyer — CHANGES_REQUESTED** [review]
One of the three disabled tests has had the linked PR merged, but the other two have not. I'm not sure why the tests are all passing without that - I'll look into it.

(I do love the way that this automation drops the disabled comment when everything is fine again).

**Comment by tonyandrewmeyer** [comment]
With this:

```diff
diff --git a/requirements.txt b/requirements.txt
index 29ba811..ea137a2 100644
--- a/requirements.txt
+++ b/requirements.txt
@@ -1,7 +1,8 @@
 cosl
 cryptography
 jsonschema
-ops
+#ops
+git+https://github.com/canonical/operator
 pyaml
 requests
 lightkube >= 0.11
```

I get this locally, with `1bd9e7fd96a9f5a91443e5932667c58a92c1ce5e`:

```
============================ 24 failed, 114 passed, 4 warnings in 3.94s ============================
```

**Comment by tonyandrewmeyer** [comment]
> (I do love the way that this automation drops the disabled comment when everything is fine again).

Ah, actually this is the problem when I look more closely. It's not removing the disabled comment, it's removing all the comments which also removes some of the repos to check against, and that's why things are passing.

File: .github/workflows/db-charm-tests.yaml [hunk 1]
```diff
@@ -15,15 +15,11 @@ jobs:
...
           - charm-repo: canonical/postgresql-operator
-            commit: 01cfd3e268ecd5024ded6bb6aede43ff85417b78  # 2024-09-11T16:07:20Z
+            commit: 7219d1ed1a118abbfca23aeacf3b7bf67e81a937  # rev480 rev479 2024-09-23T16:12:27Z
           - charm-repo: canonical/postgresql-k8s-operator
-            commit: f203e4d9d3ab9856b5dc0786217fce8428384e90  # 2024-09-16T12:00:58Z
-          # Waiting for an upstream PR:
-          # https://github.com/canonical/mysql-operator/pull/519
-          #- charm-repo: canonical/mysql-operator
-          #  commit: 4ff2dc867a1586d43a93d0017cc648d591fc9886  # 2024-09-13T15:38:12Z
+            commit: 5c2ae046546a7d163b9d040d6a12d1abc0ca1f70  # rev398 rev397 2024-09-24T19:23:37Z
```
Comment (tonyandrewmeyer) [inline .github/workflows/db-charm-tests.yaml:20]:
```suggestion
            commit: 5c2ae046546a7d163b9d040d6a12d1abc0ca1f70  # rev398 rev397 2024-09-24T19:23:37Z
          - charm-repo: canonical/mysql-operator
            commit: 4ff2dc867a1586d43a93d0017cc648d591fc9886  # 2024-09-13T15:38:12Z
```

File: .github/workflows/observability-charm-tests.yaml [hunk 2]
```diff
@@ -16,15 +16,7 @@ jobs:
...
-            commit: 4b2f567c43bc67e9d1fe24f018ba186dcd0396ed  # 2024-08-28T11:32:28Z
-          # Waiting for an upstream PR:
-          # https://github.com/canonical/prometheus-k8s-operator/pull/639
-          #- charm-repo: canonical/prometheus-k8s-operator
-          #  commit: 7518ec8343941ca91248614fd8f2d50fdd9e068c  # rev135 2024-09-06T12:10:02Z
-          # Waiting for an upstream PR:
-          # https://github.com/canonical/grafana-k8s-operator/pull/347
-          #- charm-repo: canonical/grafana-k8s-operator
-          #  commit: 1bd9e7fd96a9f5a91443e5932667c58a92c1ce5e  # 2024-09-03T14:57:42Z
+            commit: 7518ec8343941ca91248614fd8f2d50fdd9e068c  # rev135 2024-09-06T12:10:02Z
```
Comment (tonyandrewmeyer) [inline .github/workflows/observability-charm-tests.yaml:19]:
```suggestion
            commit: 7518ec8343941ca91248614fd8f2d50fdd9e068c  # rev135 2024-09-06T12:10:02Z
          # Waiting for an upstream PR:
          # https://github.com/canonical/prometheus-k8s-operator/pull/639
          #- charm-repo: canonical/prometheus-k8s-operator
          #  commit: 7518ec8343941ca91248614fd8f2d50fdd9e068c  # rev135 2024-09-06T12:10:02Z
          # Waiting for an upstream PR:
          # https://github.com/canonical/grafana-k8s-operator/pull/347
          #- charm-repo: canonical/grafana-k8s-operator
          #  commit: 1bd9e7fd96a9f5a91443e5932667c58a92c1ce5e  # 2024-09-03T14:57:42Z
```

**Review by tonyandrewmeyer — CHANGES_REQUESTED** [review]

**Comment by dimaqq** [comment]
> > (I do love the way that this automation drops the disabled comment when everything is fine again).
> 
> Ah, actually this is the problem when I look more closely. It's not removing the disabled comment, it's removing all the comments which also removes some of the repos to check against, and that's why things are passing.

Ahhh!!! maybe a solution is to have separate comment nodes?
I tried to be round-trip safe with yaml, but there's only so much I can do.

**Review by dimaqq — APPROVED** [review]

File: .github/workflows/db-charm-tests.yaml [hunk 3]
```diff
@@ -15,15 +15,13 @@ jobs:
...
+          - charm-repo: canonical/mysql-operator
+            commit: 4ff2dc867a1586d43a93d0017cc648d591fc9886  # 2024-09-13T15:38:12Z
           - charm-repo: canonical/mysql-k8s-operator
-            commit: 2009919583eddad22656b9b4ebe95adbcd518c0f  # rev199 rev198 2024-09-16T18:14:16Z
+            commit: 61db833ba8318773de565d7d9b56031baf3e75f7  # 2024-09-19T11:48:20Z
```
Comment (tonyandrewmeyer) [inline .github/workflows/db-charm-tests.yaml:24]:
```suggestion
            commit: 9fe60aa9f0a00636e06902d0849d36e1e1525df2  # 2024-09-22
```

File: .github/workflows/db-charm-tests.yaml [hunk 4]
```diff
@@ -15,15 +15,13 @@ jobs:
...
-          # Waiting for an upstream PR:
-          # https://github.com/canonical/mysql-operator/pull/519
-          #- charm-repo: canonical/mysql-operator
-          #  commit: 4ff2dc867a1586d43a93d0017cc648d591fc9886  # 2024-09-13T15:38:12Z
+            commit: 5c2ae046546a7d163b9d040d6a12d1abc0ca1f70  # rev398 rev397 2024-09-24T19:23:37Z
+          - charm-repo: canonical/mysql-operator
+            commit: 4ff2dc867a1586d43a93d0017cc648d591fc9886  # 2024-09-13T15:38:12Z
           - charm-repo: canonical/mysql-k8s-operator
-            commit: 2009919583eddad22656b9b4ebe95adbcd518c0f  # rev199 rev198 2024-09-16T18:14:16Z
+            commit: 9fe60aa9f0a00636e06902d0849d36e1e1525df2  # 2024-09-22
```
Comment (tonyandrewmeyer) [inline .github/workflows/db-charm-tests.yaml:24]:
```suggestion
            commit: 61db833ba8318773de565d7d9b56031baf3e75f7  # 2024-09-19T11:48:20Z
```

File: .github/workflows/db-charm-tests.yaml [hunk 5]
```diff
@@ -15,15 +15,13 @@ jobs:
...
           - charm-repo: canonical/postgresql-operator
-            commit: 01cfd3e268ecd5024ded6bb6aede43ff85417b78  # 2024-09-11T16:07:20Z
+            commit: 7219d1ed1a118abbfca23aeacf3b7bf67e81a937  # rev480 rev479 2024-09-23T16:12:27Z
           - charm-repo: canonical/postgresql-k8s-operator
-            commit: f203e4d9d3ab9856b5dc0786217fce8428384e90  # 2024-09-16T12:00:58Z
-          # Waiting for an upstream PR:
-          # https://github.com/canonical/mysql-operator/pull/519
-          #- charm-repo: canonical/mysql-operator
-          #  commit: 4ff2dc867a1586d43a93d0017cc648d591fc9886  # 2024-09-13T15:38:12Z
+            commit: 5c2ae046546a7d163b9d040d6a12d1abc0ca1f70  # rev398 rev397 2024-09-24T19:23:37Z
```
Comment (tonyandrewmeyer) [inline .github/workflows/db-charm-tests.yaml:20]:
```suggestion
            commit: 9fe60aa9f0a00636e06902d0849d36e1e1525df2  # 2024-09-22
```

File: .github/workflows/db-charm-tests.yaml [hunk 6]
```diff
@@ -15,15 +15,13 @@ jobs:
...
           - charm-repo: canonical/postgresql-operator
-            commit: 01cfd3e268ecd5024ded6bb6aede43ff85417b78  # 2024-09-11T16:07:20Z
+            commit: 7219d1ed1a118abbfca23aeacf3b7bf67e81a937  # rev480 rev479 2024-09-23T16:12:27Z
           - charm-repo: canonical/postgresql-k8s-operator
-            commit: f203e4d9d3ab9856b5dc0786217fce8428384e90  # 2024-09-16T12:00:58Z
-          # Waiting for an upstream PR:
-          # https://github.com/canonical/mysql-operator/pull/519
-          #- charm-repo: canonical/mysql-operator
-          #  commit: 4ff2dc867a1586d43a93d0017cc648d591fc9886  # 2024-09-13T15:38:12Z
+            commit: 9fe60aa9f0a00636e06902d0849d36e1e1525df2  # 2024-09-22
```
Comment (tonyandrewmeyer) [inline .github/workflows/db-charm-tests.yaml:20]:
```suggestion
            commit: 5c2ae046546a7d163b9d040d6a12d1abc0ca1f70  # rev398 rev397 2024-09-24T19:23:37Z
```

File: .github/workflows/db-charm-tests.yaml [hunk 7]
```diff
@@ -15,15 +15,13 @@ jobs:
...
+          - charm-repo: canonical/mysql-operator
+            commit: 4ff2dc867a1586d43a93d0017cc648d591fc9886  # 2024-09-13T15:38:12Z
```
Comment (tonyandrewmeyer) [inline .github/workflows/db-charm-tests.yaml:22]:
```suggestion
            commit: 9fe60aa9f0a00636e06902d0849d36e1e1525df2  # 2024-09-22
```

**Review by tonyandrewmeyer — APPROVED** [review]

**Pull request merged**